### PR TITLE
rygel: make gtk support optional

### PR DESCRIPTION
**Pull request by jwillikers**
This allows running Rygel as a headless service without all of the extra dependencies.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ry/rygel/package.nix [hunk 1]
```diff
@@ -67,13 +69,14 @@ stdenv.mkDerivation (finalAttrs: {
...
     libsoup_3
-    gtk3
     libmediaart
     sqlite
     systemd
     tinysparql
     shared-mime-info
-  ] ++ (with gst_all_1; [
+  ]
+  ++ lib.optional withGtk gtk3
```
Comment (jtojnar) [inline pkgs/by-name/ry/rygel/package.nix:78]: Please use the following optionals is confusing to people and not extensible:

```suggestion
  ++ lib.optionals withGtk [ gtk3 ]
```

Comment (jwillikers) [reply pkgs/by-name/ry/rygel/package.nix]: Thanks, that really helps to know. Done!

File: pkgs/by-name/ry/rygel/package.nix [hunk 2]
```diff
@@ -76,11 +81,13 @@ stdenv.mkDerivation (finalAttrs: {
       gtk3
       libmediaart
       pipewire
+      libX11
```
Comment (jtojnar) [inline pkgs/by-name/ry/rygel/package.nix:84]: Why is this needed when GUI is not built?

File: pkgs/by-name/ry/rygel/package.nix [hunk 3]
```diff
@@ -76,11 +81,13 @@ stdenv.mkDerivation (finalAttrs: {
       gtk3
```
Comment (jtojnar) [inline pkgs/by-name/ry/rygel/package.nix:81]: Actually, you still have GTK here.

Comment (jwillikers) [reply pkgs/by-name/ry/rygel/package.nix:81]: Nice catch, should have removed that.

File: pkgs/by-name/ry/rygel/package.nix (hunk 2 above)
Comment (jwillikers) [reply pkgs/by-name/ry/rygel/package.nix:84]: An X11 function is explicitly called in the app unfortunately. See [this issue](https://gitlab.gnome.org/GNOME/rygel/-/issues/221). I've opened [a PR](https://gitlab.gnome.org/GNOME/rygel/-/merge_requests/27) that will hopefully allow removing the X11 dependency entirely.

File: pkgs/by-name/ry/rygel/package.nix (hunk 3 above)
Comment (Aleksanaa) [reply pkgs/by-name/ry/rygel/package.nix:81]: https://github.com/NixOS/nixpkgs/pull/362483

Comment (jwillikers) [reply pkgs/by-name/ry/rygel/package.nix:81]: I've opened #362484 to fix this.

File: pkgs/by-name/ry/rygel/package.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/by-name/ry/rygel/package.nix:84]: Could you please also add a comment linking to that next to the libx11 dependency?

Comment (jwillikers) [reply pkgs/by-name/ry/rygel/package.nix:84]: @Aleksanaa Want to do that in your PR?
